components/host-dashboard/reports-section/HostDownloadsSection.js

```diff
@@ -0,0 +1,155 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useQuery } from '@apollo/client';
+import { FormattedMessage } from 'react-intl';
+import styled from 'styled-components';
+
+import { fetchCSVFileFromRESTService } from '../../../lib/api';
+import dayjs from '../../../lib/dayjs';
+import { API_V2_CONTEXT, gqlV2 } from '../../../lib/graphql/helpers';
+import { useAsyncCall } from '../../../lib/hooks/useAsyncCall';
+
+import PeriodFilter from '../../budget/filters/PeriodFilter';
+import CollectivePickerAsync from '../../CollectivePickerAsync';
+import Container from '../../Container';
+import { Box, Flex, Grid } from '../../Grid';
+import Link from '../../Link';
+import { PERIOD_FILTER_PRESETS } from '../../PeriodFilterPresetsSelect';
+import StyledButton from '../../StyledButton';
+import StyledInputField from '../../StyledInputField';
+
+const getHostReportURL = (hostSlug, params) => {
+  const { from, to, accountsSlugs, format = 'txt' } = params || {};
+  const url = new URL(`${process.env.REST_URL}/v2/${hostSlug}/hostTransactions.${format}`);
+  url.searchParams.set('reportType', 'hostTransactions');
+
+  if (from) {
+    url.searchParams.set('dateFrom', from);
+  }
+  if (to) {
+    url.searchParams.set('dateTo', to);
+  }
+  if (accountsSlugs?.length) {
+    url.searchParams.set('hostedAccount', accountsSlugs.join(','));
+  }
+
+  return url.toString();
+};
+
+const hostReportDownloadsSectionQuery = gqlV2/* GraphQL */ `
+  query HostReportDownloadsSection($hostSlug: String!) {
+    host(slug: $hostSlug) {
+      id
+      legacyId
+    }
+  }
+`;
```

Comment:
Since we have it in the parent, Apollo should not trigger any additional query (this will be fetched from the cache). But it could indeed be simpler to just take it as a parameter.